### ISSUE TITLE
CI: Provision failure: error checking context: 'file ('/home/vagrant/go/src/github.com/cilium/cilium/test/policy_bae469ca.json')

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#7676]
```
00:07:29.035      k8s1-1.14: f1341c205ca6: Pushed
00:07:29.837      k8s1-1.14: b57c79f4a9f3: Pushed
00:07:29.837      k8s1-1.14: d60e01b37e74: Pushed
00:07:29.837      k8s1-1.14: can't load package: package ./test/bpf: C source files not allowed when not using cgo or SWIG: bpf-event-test.c elf-demo.c unit-test.c
00:07:29.837      k8s1-1.14: e45cfbc98a50: Pushed
00:07:30.639      k8s1-1.14: docker build --build-arg LOCKDEBUG=1 -f cilium-operator.Dockerfile -t "cilium/operator:"latest"" .
00:07:40.793      k8s1-1.14: error checking context: 'file ('/home/vagrant/go/src/github.com/cilium/cilium/test/policy_bae469ca.json') not found or excluded by .dockerignore'.
00:07:40.793      k8s1-1.14: Makefile:187: recipe for target 'docker-operator-image' failed
00:07:40.793      k8s1-1.14: make: *** [docker-operator-image] Error 1
00:07:46.582      k8s1-1.14: dd98abcfbc79: 
00:07:46.582      k8s1-1.14: Pushed
00:07:46.582      k8s1-1.14: 762d8e1a6054: 
00:07:46.582      k8s1-1.14: Pushed
00:08:01.238      k8s1-1.14: f3dbe1d3279e: 
00:08:01.238      k8s1-1.14: Pushed
00:08:01.238      k8s1-1.14: latest: digest: sha256:51af0c7f19d3c8e180920d9d5a8c253a0c7d6a736cb35858926a64f49a137194 size: 3665
00:08:01.238  The SSH command responded with a non-zero exit status. Vagrant
00:08:01.238  assumes that this means the command failed. The output for this command
00:08:01.238  should be in the log above. Please read the output to determine what
00:08:01.238  went wrong.
```

https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/2686/

[test_results_master_2686_BDD-Test-PR.zip](https://github.com/cilium/cilium/files/3067961/test_results_master_2686_BDD-Test-PR.zip)

**Comment by eloycoto** [comment]
Should be part of the dockerignore files. 

https://github.com/cilium/cilium/blob/a078bb313112a801a39234f9fafa610475d77359/.dockerignore#L54-L62

**Comment by ianvernon** [comment]
It looks like as if this is a JSON file that is generated by the tests ? So maybe dockerignore ignores files that were there are the start of the build, but can still pick one up that is put in the `test/` directly after the docker build has began?

From the link @tgraf provided:

```
07:48:30      k8s1-1.14: docker build --build-arg LOCKDEBUG=1 -f cilium-operator.Dockerfile -t "cilium/operator:"latest"" 
07:48:41      k8s1-1.14: error checking context: 'file ('/home/vagrant/go/src/github.com/cilium/cilium/test/policy_bae469ca.json') not found or excluded by .dockerignore'.
07:48:41      k8s1-1.14: Makefile:187: recipe for target 'docker-operator-image' failed
07:48:41      k8s1-1.14: make: *** [docker-operator-image] Error 1
```

Failure occurred at 07:48:41.

The runtime tests created the JSON file that caused the docker build to fail at that exact time:

```
07:48:39  RuntimeChaos 
07:48:39    Delete event on KVStore with CIDR identities
07:48:39    /home/jenkins/workspace/cilium-ginkgo_cilium_master/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:410
07:48:39  STEP: Installing CIDR policy
07:48:39  STEP: Setting up policy: /home/vagrant/go/src/github.com/cilium/cilium/test/policy_bae469ca.json
07:48:40  === Test Finished at 2019-04-11T07:48:39Z====
07:48:40  <Checks>
07:48:40  Number of "context deadline exceeded" in logs: 0
07:48:40  Number of "level=error" in logs: 0
07:48:40  Number of "level=warning" in logs: 0
07:48:40  Number of "Cilium API handler panicked" in logs: 0
07:48:40  Number of "Goroutine took lock for more than" in logs: 0
07:48:40  No errors/warnings found in logs
```

**Comment by tgraf** [comment]
Similar issue in:
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/6577/execution/node/88/log/

```
00:04:16.860      k8s1-1.14: echo "1a370790e 2019-04-10T23:04:25+02:00" >GIT_VERSION
00:04:16.860      k8s1-1.14: docker build \
00:04:16.860      k8s1-1.14: 	--build-arg LOCKDEBUG=1 \
00:04:16.860      k8s1-1.14: 	--build-arg V= \
00:04:16.860      k8s1-1.14: 	--build-arg LIBNETWORK_PLUGIN= \
00:04:16.860      k8s1-1.14: 	-t "cilium/cilium:"latest"" .
00:04:19.451      k8s1-1.14: error checking context: 'file ('/home/vagrant/go/src/github.com/cilium/cilium/bpf/bpf_xdp-b1a3cf1f.ll.tmp') not found or excluded by .dockerignore'.
00:04:19.451      k8s1-1.14: Makefile:169: recipe for target 'docker-image-no-clean' failed
00:04:19.451      k8s1-1.14: make: *** [docker-image-no-clean] Error 1
```

**Comment by tgraf** [comment]
It looks like we are compiling in the same directory in parallel as invoking `docker build` and running some tests.

**Comment by ianvernon** [comment]
@tgraf yes, this is because we are running the Docker tests at the same time as we are building the Docker image for the K8s tests.

**Comment by joestringer** [comment]
Hit again in #7744:

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/11639/execution/node/81/log/

**Comment by eloycoto** [comment]
Hi!

I was debugging this one for a few days, and it's a bit tricky, the main issue,
each parallel has the same folder, so the files are in there can be
modified/added by anoother process under parallel, there are two ways to attack
this problem:

**1- Jenkins to use a different folder for each parallel.**

I tried that, because is the best way that we can have, and will fix multiple
issues. I tried to use a different workspace for each parallel, but data is not
copied and execute `cp` or `rsync` makes me scary.

The main thing that I tried is this
https://jenkins.io/doc/pipeline/steps/workflow-durable-task-step/#code-ws-code-allocate-workspace

So I almost read all Jenkins doc and I do not found a way to tackle this done.

**2- Using Vagrant sshfs plugin**

Instead of using VBOXfs use rsync, as @tgraf suggested in the past, that will
make that the files per each Cilium will be hosted in the native filesystem,
that will work ok, due to no shared files between parallel stages, but make
things harder. Instead of using `rsync` in the Vagrantfile, I would use
`vagrant-sshfs` plugin, it's cleaner and sync automatic with rsync hacks.

https://github.com/dustymabe/vagrant-sshfs


**About why the dockerignore is not working**

Dockerignore should ignore that file, and I made the test and it's correctly
ignored, so the issue IMHO in what I read is that the XXX_policy.json is created
in the runtime VM, so the docker file does not have access at all to that file,
so can't stat that and make docker to fail, so can't even ignore it.  (I have no
proof, but due to multiple issues that I read I think that this one is the
behaviour)

Another reason why this flake has been seen over the last days is that we start
to build operator in CI, and the second container build takes more time and
Runtime test already started to make new policies, so due is executed later the
issue happens more often.


So, in my opinion the second option need to be follow, I do not really liked but
is the best solution to avoid this flakes, but implies that users will need to
use a Vagrantplugin (it is straigth forward), and change a little bit how the CI
boxes are provisioned.

Thanks for all!
